capi/src/wrapper.py

```diff
@@ -0,0 +1,14 @@
+# This file is part of
+from cqibolab import ffi
+
+from qibolab import execute_qasm as py_execute_qasm
+
+
+@ffi.def_extern()
+def execute_qasm(circuit, platform, nshots):
+    """Generate samples for a given circuit qasm, platform and number of
+    shots."""
+    py_circuit = ffi.string(circuit).decode("utf-8")
+    py_platform = ffi.string(platform).decode("utf-8")
+    qasm_res = py_execute_qasm(circuit=py_circuit, platform=py_platform, nshots=nshots)
+    return ffi.cast("int*", ffi.from_buffer(qasm_res.samples().ravel()))
```

Comment:
As @BrunoLiegiBastonLiegi pointed out to me, flattening here may be a bit confusing if more than one qubits are measured, but I guess it is hard to pass the matrix structure to C.

Reply:
This we discussed with @scarrazza, and we were aware of the issue.

However, in C++ there are no matrices in STL, and the vector of vectors (or array of arrays) construction it is not necessarily the most practical one.
There are libraries defining matrices and tensors in C++, but you would have to commit to a single one (with all the issue in C++ dependencies management).

The solution has been to just pass the flattened array, and delegate the reshape in their favorite matrix container to the users.

Reply:
Indeed, we could go for more complex solutions, e.g. pybind11, but probably later, if someone really uses this.